fs/artifact_fetcher.go

```diff
@@ -72,26 +77,46 @@ func newRemoteStore(refspec reference.Spec) (*remote.Repository, error) {
 		return nil, fmt.Errorf("cannot create repository %s: %w", refspec.Locator, err)
 	}
 
-	authClient := auth.DefaultClient
-	authClient.Cache = auth.DefaultCache
-	authClient.Credential = func(_ context.Context, host string) (auth.Credential, error) {
-		username, secret, err := dockerconfig.DockerCreds(host)
-		if err != nil {
-			return auth.EmptyCredential, err
-		}
-		if username == "" && secret != "" {
+	rhttpClient := rhttp.NewClient()
+
+	// set the dial timeout and response header timeout to 1 second
+	innerTransport := rhttpClient.HTTPClient.Transport
+	if t, ok := innerTransport.(*http.Transport); ok {
+		t.DialContext = (&net.Dialer{
```

Comment:
Can you make this same change in `service/resolver/registry.go`? I think that's too aggressive as well.